src/perform_step/SROCK_perform_step.jl

```diff
@@ -332,8 +332,12 @@ end
 
   sqrt_dt   = sqrt(abs(dt))
   if gen_prob
-    vec_χ .= 1//2 .+ oftype(W.dW, rand(W.rng, length(W.dW)))
-    @.. vec_χ = 2*floor(vec_χ) - 1
+    if hasfield(W, :rng)
+        rand!(W.rng, vec_χ)
+    else
+        rand!(vec_χ)
```

Comment:
for the noise wrapper it should just forward it to `W.W.rng`

Reply:
I guess its  W.source.rng